### PR TITLE
Stricter preferences, notably missingNonNullByDefaultAnnotation as error

### PR DESCRIPTION
**Pull request by vorburger**

### CODE REVIEW

**Comment by vorburger** [comment]
@sylvainlaurent @maggu2810 OK for you?

**Comment by maggu2810** [comment]
Don't know what you are doing here.
I added the different possible nullness annotations to my IDE settings, so it knows about non-null and null annotations of differenent projects (JDT, javax, Checker Franework, ...). On the Checker Franework manual page there is a summary of that ones.

File: examples/maven/jdt-ecj-settings/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations=enabled
-org.eclipse.jdt.core.compiler.annotation.missingNonNullByDefaultAnnotation=warning
+org.eclipse.jdt.core.compiler.annotation.missingNonNullByDefaultAnnotation=error
 org.eclipse.jdt.core.compiler.annotation.nonnull=org.eclipse.jdt.annotation.NonNull
 org.eclipse.jdt.core.compiler.annotation.nonnull.secondary=javax.annotation.NonNull
```
Comment (sylvainlaurent) [inline examples/maven/jdt-ecj-settings/org.eclipse.jdt.core.prefs]: BTW it's javax.annotation.Nonnull, not javax.annotation.NonNull

**Comment by sylvainlaurent** [comment]
added a README

File: examples/maven/jdt-ecj-settings/org.eclipse.jdt.core.prefs (hunk 1 above)
Comment (vorburger) [reply examples/maven/jdt-ecj-settings/org.eclipse.jdt.core.prefs]: @sylvainlaurent wow - thanks a lot for spotting this!!

**Pull request merged**